substrate/frame/system/src/migrations/mod.rs

```diff
@@ -119,3 +120,10 @@ pub fn migrate_from_dual_to_triple_ref_count<V: V2ToV3, T: Config>() -> Weight {
 	<UpgradedToTripleRefCount<T>>::put(true);
 	Weight::MAX
 }
+
+/// Migrate from `Events` to `EventSegments` when `T::EventSegmentSize > 0`
+pub fn migrate_from_events_to_event_segments<T: Config>() -> Weight {
```

Comment:
I can’t see where this migration is called from. Are we adding the migration in substrate or subspace code?

Reply:
Right, the migration will be used by the downstream projects (e.g. subspace) in their runtime, just like the storage migration we have implemented.